derivation.nix

```diff
@@ -70,15 +107,35 @@ in
           buildPhase = ''
             runHook preBuild
 
-            pnpm run ${script}
+            ${if isWorkspace
+              then ''
+                ${forEachComponent (component: ''
+                  (cd "${component}" && pnpm run ${script})
+                '')}
+              ''
+              else ''
+                pnpm run ${script}
+              ''
+            }
 
             runHook postBuild
           '';
 
           installPhase = ''
             runHook preInstall
 
-            ${if distDir == "." then "cp -r" else "mv"} ${distDir} $out
+            ${if isWorkspace
```

Comment:
It would not remove this test, because the behavior is necessarily different from a workspace with a single component.

Reply:
I instead made a different flag that controls this specific behavior. I think this is cleaner.